### DISCUSSION TITLE
第二次调用findCachedClient是否多余?

### DISCUSSION THREAD

**Discussion by carlos19960601** [smallnest/rpcx#726] [Q&A]
在这段逻辑中，第二次调用findCachedClient是否多余？
在什么情况下第1次调用findCachedClient时client == nil，而第2次调用的时候client != nil呢？

https://github.com/smallnest/rpcx/blob/6b17ab253a936280f93069b84b6409e45d8ed54f/client/xclient.go#L290-L301

**Comment by smallnest** [comment]
第一次选出的client正在被关闭
